File: the_old_switcheroo.js
function vowel2index(str) {
    let count1 = 0;
    let regex = /[aeiou]/gi
    let strArr = [...str]
    let newStr = []
    strArr.forEach((char)=>{
      if(char.match(regex)){
        newStr.push(count1+1)
      }
      if(!char.match(regex)){
        newStr.push(char)
      }
      count1++
    });
    return newStr.join('')
  }

  console.log(vowel2index("this is my string"))
  console.log(vowel2index("Codewars is the best site in the world"))
  console.log(vowel2index("Tomorrow is going to be raining"))
  console.log(vowel2index(''))

File: the_Old_Switcheroo.py
import re
def encode(string):
    alphabet_low = list('abcdefghijklmnopqrstuvwxyz')
    alphabet_up = list('ABCDEFGHIJKLMNOPQRSTUVWXYZ')
    char_lst = list(string)
    new_string = ''
    for char in char_lst:
        if re.match(r"[a-z]", char):
            num = alphabet_low.index(char)
            new_string += str(num + 1)
        if re.match(r"[A-Z]", char):
            num = alphabet_up.index(char)
            new_string += str(num + 1)
        if re.match(r"[\W\[\]\d]", char):
            new_string += char
    return new_string


print(encode("this is my string"))
print(encode("Codewars is the best site in the world"))
print(encode("Tomorrow is going to be raining"))
print(encode(''))